src/Sulu/Bundle/AdminBundle/Resources/js/containers/Application/Application.js

```diff
@@ -15,22 +15,68 @@ import {Backdrop} from '../../components';
 import Login from '../Login';
 import applicationStyles from './application.scss';
 
+const USER_SETTING_PREFIX = 'sulu_admin.application';
+const USER_SETTING_NAVIGATION_PINNED = 'navigation_pinned';
+
+function getNavigationPinnedSettingKey() {
+    return USER_SETTING_PREFIX + '.' + USER_SETTING_NAVIGATION_PINNED;
+}
+
 type Props = {
     router: Router,
 };
 
 @observer
 export default class Application extends React.Component<Props> {
     @observable navigationVisible: boolean = false;
+    @observable navigationPinned: boolean = false;
+
+    navigationPinnedDisposer: () => void;
+
+    constructor() {
+        super();
+
+        this.pinNavigation(!!userStore.getPersistentSetting(getNavigationPinnedSettingKey()));
+
+        this.navigationPinnedDisposer = autorun(
+            () => userStore.setPersistentSetting(getNavigationPinnedSettingKey(), this.navigationPinned ? 1 : 0)
+        );
+    }
 
     @action toggleNavigation() {
+        if (this.navigationPinned) {
+            return;
+        }
+
         this.navigationVisible = !this.navigationVisible;
     }
 
+    @action pinNavigation(value?: boolean) {
+        if (value !== undefined) {
+            this.navigationPinned = value;
+        } else {
+            this.navigationPinned = !this.navigationPinned;
+        }
+
+        if (this.navigationPinned) {
+            this.navigationVisible = true;
+        } else {
+            this.navigationVisible = false;
+        }
+    }
+
+    componentWillUnmount() {
+        this.navigationPinnedDisposer();
+    }
+
     handleNavigationButtonClick = () => {
         this.toggleNavigation();
     };
 
+    handleNavigationPin = () => {
+        this.pinNavigation();
```

Comment:
Is it a good idea to call this function without an argument? What if you pass a `pinned` parameter to the `handleNavigationPin` callback, and then call the correct `pinNavigation` or `unpinNavigation` method in here?